grpc/nodejs/BUILD

```diff
@@ -16,21 +16,72 @@
 #
 
 package(default_visibility = ["//visibility:public"])
-load("@stackb_rules_proto//node:node_grpc_compile.bzl", "node_grpc_compile")
 load("@graknlabs_dependencies//tool/checkstyle:rules.bzl", "checkstyle_test")
+load("@build_bazel_rules_nodejs//:index.bzl", "pkg_npm")
+load("@graknlabs_bazel_distribution//npm:rules.bzl", "assemble_npm", "deploy_npm")
+load("@graknlabs_dependencies//distribution:deployment.bzl", "deployment")
+load("//:deployment.bzl", github_deployment = "deployment")
 
-node_grpc_compile(
+genrule(
     name = "protocol",
-    deps = [
-        "//protobuf:answer-proto",
-        "//protobuf:concept-proto",
-        "//protobuf:database-proto",
-        "//protobuf:grakn-proto",
-        "//protobuf:options-proto",
-        "//protobuf:query-proto",
-        "//protobuf:session-proto",
-        "//protobuf:transaction-proto",
-    ]
+    outs = [
+        "protobuf/answer_pb.d.ts",
+        "protobuf/answer_pb.js",
+        "protobuf/concept_pb.d.ts",
+        "protobuf/concept_pb.js",
+        "protobuf/database_pb.d.ts",
+        "protobuf/database_pb.js",
+        "protobuf/grakn_pb.d.ts",
+        "protobuf/grakn_pb.js",
+        "protobuf/grakn_grpc_pb.js",
+        "protobuf/grakn_grpc_pb.d.ts",
+        "protobuf/options_pb.d.ts",
+        "protobuf/options_pb.js",
+        "protobuf/query_pb.d.ts",
+        "protobuf/query_pb.js",
+        "protobuf/session_pb.d.ts",
+        "protobuf/session_pb.js",
+        "protobuf/transaction_pb.d.ts",
+        "protobuf/transaction_pb.js",
+    ],
+    cmd = "ln -s $(execpath @npm//grpc_tools_node_protoc_ts/bin:protoc-gen-ts).runfiles/npm/node_modules ./node_modules ;\
+        $(execpath @npm//grpc-tools/bin:grpc_tools_node_protoc) \
+            --plugin='protoc-gen-ts=$(rootpath @npm//:node_modules/grpc_tools_node_protoc_ts/bin/protoc-gen-ts)' \
+            --js_out='import_style=commonjs,binary:./$(@D)/' \
+            --grpc_out='grpc_js:./$(@D)/' \
+            --ts_out='grpc_js:./$(@D)/' \
+            --proto_path=`dirname $(execpath //:WORKSPACE)` \
+            $(execpaths //protobuf:proto-raw-buffers);",
+        tools = [
+            "@npm//grpc-tools/bin:grpc_tools_node_protoc",
+            "@npm//grpc_tools_node_protoc_ts/bin:protoc-gen-ts",
+            "@npm//:node_modules/grpc_tools_node_protoc_ts/bin/protoc-gen-ts",
+            "@npm//grpc_tools_node_protoc_ts",
+            "@npm//google-protobuf",
+            "//protobuf:proto-raw-buffers",
+            "//:WORKSPACE"
+        ],
+)
+
+pkg_npm(
+    name = "protocol-package",
+    package_name = "graknlabs_protocol",
+    srcs = glob([
+        "package.json",
+    ]),
+    deps = [":protocol"],
+)
+
+assemble_npm(
+    name = "assemble-npm",
+    target = ":protocol-package",
+)
+
+deploy_npm(
+    name = "deploy-npm",
+    target = ":assemble-npm",
+    snapshot = deployment["npm.snapshot"],
+    release = deployment["npm.release"],
 )
```

Comment:
Compile, assemble, package and deploy the npm package, using the same mechanisms as client-nodejs